crates/ika-core/src/dwallet_mpc/mpc_session.rs

```diff
@@ -1,1340 +1,294 @@
-                is_ready: true,
-                malicious_parties: vec![],
-            })
-        } else {
-            info!(
-                ?self.consensus_rounds_since_quorum_reached,
-                ?self.current_round,
-                ?self.agreed_mpc_protocol,
-                messages_count_for_current_round=?self.serialized_full_messages.get(&(self.current_round - 1)).unwrap_or(&HashMap::new()).len(),
-                "Quorum reached for MPC session but delay not passed yet, waiting for another round",
-            );
-            self.consensus_rounds_since_quorum_reached += 1;
-            Ok(ReadyToAdvanceCheckResult {
-                is_ready: false,
-                malicious_parties: vec![],
-            })
-        }
-    }
-
-    pub(crate) fn check_quorum_for_next_crypto_round(
-        &mut self,
-    ) -> DwalletMPCResult<ReadyToAdvanceCheckResult> {
-        match self.status {
-            MPCSessionStatus::Active => {
-                // MPC First round doesn't require a threshold of messages to advance.
-                // This is the round after the MPC event.
-                let is_quorum_reached = self
-                    .weighted_threshold_access_structure
-                    .is_authorized_subset(
-                        &self
-                            .serialized_full_messages
-                            // Check if we have the threshold of messages for the previous round
-                            // to advance to the next round.
-                            .get(&(self.current_round - 1))
-                            .unwrap_or(&HashMap::new())
-                            .keys()
-                            .cloned()
-                            .collect::<HashSet<PartyID>>(),
-                    )
-                    .is_ok();
-                // Round 1 does not have a delay.
-                if self.current_round == 1 {
-                    Ok(ReadyToAdvanceCheckResult {
-                        is_ready: true,
-                        malicious_parties: vec![],
-                    })
-                } else if is_quorum_reached
-                    && self.received_more_messages_since_last_advance
-                    && self.agreed_mpc_protocol.is_some()
-                {
-                    self.wait_consensus_rounds_delay()
-                } else {
-                    Ok(ReadyToAdvanceCheckResult {
-                        is_ready: false,
-                        malicious_parties: vec![],
-                    })
-                }
+    /// - **NetworkDkg protocol**: requires delay for the third round
+    ///   using `network_dkg_third_round_delay` config.
+    /// - **DecryptionKeyReconfiguration protocol**: requires delay for the third round
+    ///   using `decryption_key_reconfiguration_third_round_delay` config.
+    /// - **Other protocols**: No delay required, always ready to advance
+    fn consensus_rounds_delay_for_mpc_round(&self) -> usize {
```

Comment:
Also here, I think this func should get mpc_event_data and not assume it's there